src/controllers/ZoomController.ts

```diff
@@ -0,0 +1,42 @@
+import { EditorAPI, Id } from '../../types/CommonTypes';
+import { getEditorResponseData } from '../utils/EditorResponseData';
+
+/**
+ * The ZoomController is responsible for all communication regarding Zoom functionality.
+ * Methods inside this controller can be called by `window.SDK.zoom.{method-name}`
+ */
+export class ZoomController {
+    /**
+     * @ignore
+     */
+    #editorAPI: EditorAPI;
+
+    /**
+     * @ignore
+     */
+    constructor(editorAPI: EditorAPI) {
+        this.#editorAPI = editorAPI;
+    }
+
+    /**
+     * This method fits the page to the given rectangle or the viewport available
+     * Optional parameters `left`, `top`, `width` and `height` are needed to define the rectangle to fit the page to
+     * If any or all of them aren't provided the page will fit the whole viewport available
+     * @param pageId The ID of a specific page
+     * @param left
+     * @param top
+     * @param width
+     * @param height
+     * @returns
+     */
+    zoomToFit = async (
+        pageId?: Id | null,
+        left?: number | null,
+        top?: number | null,
+        width?: number | null,
+        height?: number | null,
+    ) => {
+        const res = await this.#editorAPI;
+        return res.zoomToFit(pageId, left, top, width, height).then((result) => getEditorResponseData<null>(result));
+    };
+}
```

Comment:
Isn't it a bit overkill to have a zoomcontroller?
Doesn't it make sense to put all canvas related stuff under a CanvasController?
Maybe there we can make a distinction f.e.
SDK.canvas.zoom.zoomToFit?

Reply:
I'll leave @evan-mcgeek give its input but we discussed it several times and I think we agreed that there would be a discussion following the SDK PR.
We hesitated between putting it beside a zoom controller, canvas controller (or even a page controller).

I myself don't have a strong opinion, it's probably overkill for now, not sure how many zoom functions we are going to get (a second one is coming for general canvas zoom). Canvas controller looks the most generic.

_As a side-note, it's gonna be called zoomToPage._

@evan-mcgeek what do you think ?

Reply:
I'd leave it separated IMO, considering some possible additional functionality in the future

Reply:
so what would you say @brapoprod ?